docs/storage_transformers/sharding/v1.0.rst

```diff
@@ -0,0 +1,185 @@
+        {
+          "storage_transformer": "https://purl.org/zarr/spec/storage_transformers/sharding/1.0",
+          "configuration": {
+            "format": "indexed",
+            "chunks_per_shard": [
+                2,
+                2
+            ]
+          }
+      ]
+    }
+
+``format``
+
+    Specifies a `Binary shard format`_. In this version, the only binary format is the
+    ``indexed`` format.
+
+``chunks_per_shard``
+
+    An array of integers providing the number of chunks that are combined in a shard
+    for each dimension of the Zarr array, where each chunk may only start at a position
+    that is divisble by ``chunks_per_shard`` per dimension, e.g. starting at the zero-origin.
+    The length of the array must match the length of the array metadata ``shape`` entry.
+    For example, a value ``[32, 2]`` indicates that 64 chunks are combined in one shard,
+    32 along the first dimension, and for each of those 2 along the second dimension.
+    Valid starting positions for a shard in the chunk-grid are therefore ``[0, 0]``,
+    ``[32, 2]``, ``[32, 4]``, ``[64, 2]`` or ``[96, 18]``.
+
+
+Key & value transformation
+==========================
+
+The storage transformer protocol defines the abstract interface to be the same
+as the :ref:`abstract-store-interface`.
+
+The Zarr store interface is defined in terms of `keys` and `values`,
+where a `key` is a sequence of characters and a `value` is a sequence
+of bytes. A key-value pair is called `entry` in the following part.
+
+This sharding transformer only adapts entries where the key starts
+with `data/root`, as they indicate data keys for array chunks. All other
+entries are simply passed on.
+
+Entries starting with ``data/root`` are grouped by their common shard, assuming
+storage keys from a regular chunk grid which may use a customly configured
+``chunk separator``:
+For all entries that are part of the same shard the key is changed to the
+shard-key and the values are combined in the `Binary shard format`_ described
+below. The new shard-key is the chunk key divided by ``chunks_per_shard`` and
+floored per dimension. E.g. for ``chunks_per_shard=[32, 2]``, the chunk grid
+position ``[96, 18]`` (e.g. key "data/root/foo/baz/c96/18") is transformed to
+the shard grid position ``[3, 9]`` and reassigned to the respective new key,
+honoring the original chunk separator (e.g. "data/root/foo/baz/c3/9").
+Chunk grid positions ``[96, 19]``, ``[97, 18]``, …, up to ``[127, 19]`` will
+also have the same shard grid position ``[3, 9]``.
+
+
+Binary shard format
+===================
+
+The only binary format is the ``indexed`` format, as specified by the ``format``
+configuration key. Other binary formats might be added in future versions.
+
+In the indexed binary format chunks are written successively in a shard, where
+unused space between them is allowed, followed by an index referencing them.
+The index holds an `offset, length` pair of little-endian uint64 per chunk,
```

Comment:
At which offset do I find the index?

Reply:
At the end of the file, I added an explanation.